esyi/FetchStorage.ml

```diff
@@ -312,66 +339,105 @@ type installation = {
 let install ~onBeforeLifecycle dist =
   (** TODO: need to sync here so no two same tasks are running at the same time *)
   let open RunAsync.Syntax in
-  RunAsync.contextf (
 
+  let install sourceInstallPath () =
+    let sourceStagePath, run =
+      (* We are getting EACCESS error on Windows if we try to rename directory
+       * from stage to install after we read a file from there. It seems we are
+       * leaking fds and Windows prevent rename from working.
+       *
+       * For now we are unpacking and running lifecycle directly in a final
+       * directory and in case of an error we do a cleanup by removing the
+       * install directory (so that subsequent installation attempts try to do
+       * install again).
+       *)
+      match System.Platform.host with
+      | Windows ->
+        let run f =
+          let rec cleanup ?(n=3) () =
```

Comment:
Super sketchy but it seem to work, @bryphe 😬

Reply:
👍 Nice find, @andreypopp !

IMO I see this retry strategy often in Windows I/O APIs (in different frameworks). I think it's reasonable! We might even want to push it to a lower layer, like inside `Fs.rmPathLwt`.